tests/crypto_test.py

```diff
@@ -26,3 +56,286 @@ def test_returned_string_is_lower_case(self):
+        with self.assertRaises(GPGProblem) as caught:
+            crypto.validate_key(make_key(invalid=True))
+
+        self.assertEqual(caught.exception.code, GPGCode.KEY_INVALID)
+
+    def test_encrypt(self):
+        with self.assertRaises(GPGProblem) as caught:
+            crypto.validate_key(make_key(can_encrypt=False), encrypt=True)
+
+        self.assertEqual(caught.exception.code, GPGCode.KEY_CANNOT_ENCRYPT)
+
+    def test_encrypt_no_check(self):
+        try:
+            crypto.validate_key(make_key(can_encrypt=False))
+        except GPGProblem as e:
+            raise AssertionError(e)
+
+    def test_sign(self):
+        with self.assertRaises(GPGProblem) as caught:
+            crypto.validate_key(make_key(can_sign=False), sign=True)
+
+        self.assertEqual(caught.exception.code, GPGCode.KEY_CANNOT_SIGN)
+
+    def test_sign_no_check(self):
+        try:
+            crypto.validate_key(make_key(can_sign=False))
+        except GPGProblem as e:
+            raise AssertionError(e)
+
+
+class TestCheckUIDValidity(unittest.TestCase):
+
+    def test_valid_single(self):
+        key = make_key()
+        key.uids[0] = make_uid(mock.sentinel.EMAIL)
+        ret = crypto.check_uid_validity(key, mock.sentinel.EMAIL)
+        self.assertTrue(ret)
+
+    def test_valid_multiple(self):
+        key = make_key()
+        key.uids = [
+            make_uid(mock.sentinel.EMAIL),
+            make_uid(mock.sentinel.EMAIL1),
+        ]
+
+        ret = crypto.check_uid_validity(key, mock.sentinel.EMAIL1)
+        self.assertTrue(ret)
+
+    def test_invalid_email(self):
+        key = make_key()
+        key.uids[0] = make_uid(mock.sentinel.EMAIL)
+        ret = crypto.check_uid_validity(key, mock.sentinel.EMAIL1)
+        self.assertFalse(ret)
+
+    def test_invalid_revoked(self):
+        key = make_key()
+        key.uids[0] = make_uid(mock.sentinel.EMAIL, revoked=True)
+        ret = crypto.check_uid_validity(key, mock.sentinel.EMAIL)
+        self.assertFalse(ret)
+
+    def test_invalid_invalid(self):
+        key = make_key()
+        key.uids[0] = make_uid(mock.sentinel.EMAIL, invalid=True)
+        ret = crypto.check_uid_validity(key, mock.sentinel.EMAIL)
+        self.assertFalse(ret)
+
+    def test_invalid_not_enough_trust(self):
+        key = make_key()
+        key.uids[0] = make_uid(mock.sentinel.EMAIL,
+                               validity=gpgme.VALIDITY_UNDEFINED)
+        ret = crypto.check_uid_validity(key, mock.sentinel.EMAIL)
+        self.assertFalse(ret)
+
+
+class TestListKeys(_TestKey):
+
+    def test_list_no_hints(self):
+        # This only tests that you get 3 keys back (the number in our test
+        # keyring), it might be worth adding tests to check more about the keys
+        # returned
+        values = crypto.list_keys()
+        self.assertEqual(len(list(values)), 3)
+
+    def test_list_hint(self):
+        values = crypto.list_keys(hint="ambig")
+        self.assertEqual(len(list(values)), 2)
```

Comment:
It does not, I'll fix the docstring. I actually returns a `gpgme.KeyIter`, which is obviously an iterator.